### DISCUSSION TITLE
networking possible to use websockify ?

### DISCUSSION THREAD

**Discussion by bit-lang** [copy/v86#1157] [Q&A] [answered]
[benjamincburns/websockproxy](https://github.com/benjamincburns/websockproxy) is recommended in the [networking doc](https://github.com/copy/v86/blob/master/docs/networking.md)
is it possible to use [websockify](https://github.com/novnc/websockify) as the networking proxy?

**Comment by SuperMaxusa** [comment, marked as answer]
No, because v86' networking is based on [Layer 2](https://en.wikipedia.org/wiki/Data_link_layer) (ie. sends/receives an Ethernet frames to/from the proxy, and websockproxy replicates all networking base: DHCP server, DNS server etc.) while websockify is based on almost [Layer 4](https://en.wikipedia.org/wiki/Transport_layer) (from README: "_At the most basic level, websockify just translates WebSockets traffic to normal socket traffic_")

**Reply by bit-lang** [reply]
thank you for the clarification

**Comment by ProgrammerIn-wonderland** [comment]
Wisp as a protocol is similar to websockify, just multiplexed (sending multiple TCP streams in one websocket for performance reasons), and can be used now. I actually used websockify initially while writing wisp_network.js, if you remove the wisp code and create a new WebSocket() for each TCP stream it will work. It shouldn't be a big task if you're still interested

**Reply by ProgrammerIn-wonderland** [reply]
Super ancient code now, and messy as can be (this was my first prototype) but this is a modified fetch_network.js which creates a new WebSocket for each TCP stream at wss://server/hostname:port, like websockify but giving the target to the server instead of it being hard coded. It can be modified to use websockify pretty easily 

https://github.com/MercuryWorkshop/v86/blob/0dbfa1bdd1344d49b304469fdfca5935d1332908/src/browser/fetch_network.js
